crates/starknet-devnet/tests/test_get_class.rs

```diff
@@ -133,4 +133,148 @@ mod get_class_tests {
             _ => panic!("Invalid error: {err:?}"),
         }
     }
+
+    #[tokio::test]
+    async fn test_getting_class_at_various_blocks() {
+        let devnet_args = ["--state-archive-capacity", "full"];
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&devnet_args).await.unwrap();
+
+        let (signer, account_address) = devnet.get_first_predeployed_account().await;
+        let predeployed_account = Arc::new(SingleOwnerAccount::new(
+            devnet.clone_provider(),
+            signer.clone(),
+            account_address,
+            chain_id::SEPOLIA,
+            ExecutionEncoding::New,
+        ));
+
+        let (contract_class, casm_class_hash) =
+            get_events_contract_in_sierra_and_compiled_class_hash();
+
+        // declare the contract
+        let declaration_result = predeployed_account
+            .declare(Arc::new(contract_class.clone()), casm_class_hash)
+            .max_fee(FieldElement::from(1e18 as u128))
+            .send()
+            .await
+            .unwrap();
+
+        let declaration_block = devnet.get_latest_block_with_tx_hashes().await.unwrap();
+
+        // create an extra block so the declaration block is no longer the latest
+        devnet.create_block().await.unwrap();
+
+        // getting class at the following block IDs should be successful
+        let expected_class = ContractClass::Sierra(contract_class);
+        for block_id in [
+            BlockId::Tag(BlockTag::Latest),
+            BlockId::Tag(BlockTag::Pending),
+            BlockId::Number(declaration_block.block_number),
+            BlockId::Number(declaration_block.block_number + 1),
+            BlockId::Hash(declaration_block.block_hash),
+        ] {
+            let retrieved_class = devnet
+                .json_rpc_client
+                .get_class(block_id, declaration_result.class_hash)
+                .await
+                .unwrap();
+
+            assert_cairo1_classes_equal(&retrieved_class, &expected_class).unwrap();
+        }
+
+        // getting class at the following block IDs should NOT be successful
+        for block_id in [BlockId::Number(declaration_block.block_number - 1)] {
+            let retrieved =
+                devnet.json_rpc_client.get_class(block_id, declaration_result.class_hash).await;
+            match retrieved {
+                Err(ProviderError::StarknetError(StarknetError::ClassHashNotFound)) => (),
+                other => panic!("Unexpected response: {other:?}"),
+            }
+        }
+    }
+
+    #[tokio::test]
+    async fn test_getting_class_with_blocks_on_demand() {
```

Comment:
shouldn't this test be moved to crates/starknet-devnet/tests/test_blocks_generation.rs
?

Reply:
it makes sense to have it in either of those files